Dalamud/Game/Addon/Events/AddonEventManager.cs

```diff
@@ -146,11 +156,34 @@ internal void RemovePluginEventController(string pluginId)
         if (this.pluginEventControllers.FirstOrDefault(entry => entry.PluginId == pluginId) is { } controller)
         {
             Log.Verbose($"Removing PluginEventController for: {pluginId}");
-            this.pluginEventControllers.Remove(controller);
-            controller.Dispose();
+            this.oldEventControllers.Add(controller);
         }
     }
+    
+    private void OnFrameworkUpdate(IFramework framework1)
+    {
+        if (this.newEventControllers.Any())
+        {
+            foreach (var toAdd in this.newEventControllers)
+            {
+                this.pluginEventControllers.Add(toAdd);
+            }
+        
+            this.newEventControllers.Clear();  
+        }
 
+        if (this.oldEventControllers.Any())
+        {
+            foreach (var toRemove in this.oldEventControllers)
+            {
+                this.pluginEventControllers.Remove(toRemove);
+                toRemove.Dispose();
```

Comment:
This should be wrapped in `try { ... } catch { /* ignore */ }`, or [`ExplicitDisposeIgnoreExceptions`](https://github.com/goatcorp/Dalamud/blob/fb864dd56d9dbc868496f579138f50399b7acec8/Dalamud/Utility/Util.cs#L655) extension method should be used. It was fine to not catch exception from `Dispose` before moving the it into `Framework.Update` block, since the exception would not have had an effect on handling of other event controllers.

While `List<T>.Add(...)` technically can throw too, the reason for that would be OOM, at which point there's no point trying to remedy that, so I think it's alright to leave the adding part out for exception handling. On the other hand, `IDisposable.Dispose` got more random reasons to throw, so yeah.